### PR TITLE
[WIP] Run test suite with executionOrder="random"

### PR DESCRIPTION
**Pull request by Majkl578**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | #3449

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
The fact that all builds went completely haywire is quite telling: I'd love to have this merged though.

Is there any way to make sure that the random seed is displayed in the build? That's a real blocker.

**Comment by Majkl578** [comment]
@Ocramius Yes, see https://github.com/sebastianbergmann/phpunit/issues/3515 + https://github.com/sebastianbergmann/phpunit/pull/3524 - the seed will be displayed by default in the next PHPUnit release. :+1:

File: composer.lock [hunk 1]
```diff
@@ -650,7 +652,7 @@
                     "homepage": "https://nette.org/contributors"
                 }
             ],
-            "description": "💎 Nette Dependency Injection Container: Flexible, compiled and full-featured DIC with perfectly usable autowiring and support for all new PHP 7.1 features.",
+            "description": "? Nette Dependency Injection Container: Flexible, compiled and full-featured DIC with perfectly usable autowiring and support for all new PHP 7.1 features.",
```
Comment (Majkl578) [inline composer.lock]: This looks like a composer bug on Windows.

Comment (morozov) [reply composer.lock]: Did you run the update on Windows? 😱

Comment (Majkl578) [reply composer.lock]: Yes. 😹 Composer should be portable, right? 😅

**Comment by morozov** [comment]
While we are fixing the failures, let's try to document what the end result should like from the CI perspective. IIRC, we considered running PR tests with a fixed seed in order to avoid irrelevant failures. The question is, what if a PR introduces an implicit dependency? Should cron-scheduled builds and the push ones have the same order settings? Or we just go wild and enable the random order everywhere?

File: composer.lock (hunk 1 above)
Comment (morozov) [reply composer.lock]: Apparently, it's not only a Windows issue. I'm getting the same consistently on one of my two Linux machines (see #3604). Did you have a chance to look deeper or maybe report an issue? I haven't found anything related.

**Comment by morozov** [comment]
Closing in favor of https://github.com/doctrine/dbal/pull/4731. There are no more known failures in failures in random mode but there doesn't seem to be a way to allow build failures on GitHub Actions for a certain configuration.

**Pull request closed without merging**